features/distributed/kv-store/json/postgres/src/main/resources/META-INF/opennms/applicationContext-postgresJsonStore.xml

```diff
@@ -7,9 +7,9 @@
 
     <bean name="dataSource" class="org.opennms.core.db.DataSourceFactory" factory-method="getInstance"/>
 
-    <bean name="jsonStore" class="org.opennms.features.distributed.kvstore.json.postgres.PostgresJsonStore">
+    <bean name="postgresJsonStore" class="org.opennms.features.distributed.kvstore.json.postgres.PostgresJsonStore">
```

Comment:
Currently the intent is that there would only ever be one instance fo the jsonStore. Changing the name here creates an implicit dependency on this specific implementation (the postgres one). I think we should keep the naming as is since it simplifies swapping out implementations in the future.

Since there is only one instance, we do not need to be able to differentiate based on the name.

Reply:
I specifically want to use `postgresJsonStore` for the current usecase.  A `NoSQL` jsonStore would be slow for this usecase.  I think  having `postgresJsonStore` as bean name would be more visible for anybody trying to understand the code or changing the code in future instead of assuming that it may be Postgres.  That is my intention of changing this.

Reply:
The client code shouldn't be deciding which implementation it is going to use. There is only going to be one implementation per deployment right now so a qualifier is going to fail if the one you are looking for is not installed. The bean was named that way for precisely the reason that people should not be assuming that it is implemented with Postgres, that implementation detail should be irrelevant for use cases of this API.

Client code should be agnostic of implementation based on the current paradigm. We may look into providing multiple implementations in the future at which point qualifier based selection would be reasonable, but that is not the case currently.

Can you explain why you believe the Cassandra implementation would be too slow? The API only provides primary key based get/set. NoSQL DB such as Cassandra handle that case very well.

Reply:
From my limited understanding some NoSQL DBs are optimized for writing lots of data.  In case of our Newts solution, Rest APIs take longer fetching the graphs. Anyways I will go with your naming pattern.